packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -397,21 +425,49 @@ export async function azureParameterMigration(context: MigrationContext): Promis
   }
 }
 
+export async function showNotification(
+  ctx: CoreHookContext,
+  versionForMigration: VersionForMigration
+): Promise<boolean> {
+  const isNonmodalMessage = getParameterFromCxt(ctx, Parameters.isNonmodalMessage);
+  if (isNonmodalMessage) {
+    return await showNonmodalNotification(ctx, versionForMigration);
+  }
+  const confirmOnly = getParameterFromCxt(ctx, Parameters.confirmOnly);
+  if (confirmOnly) {
+    return await showConfirmOnlyNotification(ctx);
+  }
+  const skipUserConfirm = getParameterFromCxt(ctx, Parameters.skipUserConfirm);
+  if (skipUserConfirm) {
+    sendTelemetryEvent(Component.core, TelemetryEvent.ProjectMigratorNotification, {
```

Comment:
Is there any case that we need to skip confirmation?

Reply:
vs platform has a case for it.